bloom-housing/bloom
# refactor: remove business logic, strings from waitlist component

# Pull Request Template

## Issue Overview

This PR addresses #2688


## Description

Removes business logic, hardcoded strings, and styling issues from the Waitlist component. I generalized the name to `QuantityRowSection` so that it can be used outside the context of needing to refer to a waitlist.

## How Can This Be Tested/Reviewed?

[Link to the Storybook deploy preview stories
](https://deploy-preview-2689--dev-storybook-bloom.netlify.app/?path=/story/listing-sidebar-quantity-row-section--open-and-all-values-supplied)
We have a new prop set that adds the strings, and removes the `isWaitlistOpen` check, which we now check before we render the component. Additionally allows JSX content in the description.

The new style updates are: a darker header to meet color contrast, & larger and fully bolded quantity lines.

This is a breaking change to this component due to prop changes.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
